### PR TITLE
Rmsh

### PR DESCRIPTION
**Pull request by garrettwrong**
Implements `sph_harm` using the recurrence based normalized associated Legendre polynomial method.  FB3D should now operate for the same orders as FFB3D.

Adds tests that compare low orders against `scipy`. 
Checks `scipy` is not finite for higher orders (they may fix one day...).
Checks this implementation is finite for an order that was previously a problem.

Purges `pyshtools`.

### CODE REVIEW

**Comment by garrettwrong** [comment]
Closes #1169

**Comment by garrettwrong** [comment]
Another iteration on  #923

File: tests/test_basis_utils.py [hunk 1]
```diff
@@ -9,10 +10,56 @@
...
     unique_coords_nd,
 )
 
 
+def test_sph_harm_low_order():
+    """
+    Test the `sph_harm` implementation matches `scipy` at lower orders.
+    """
+    m = 3
+    j = 5
```
Comment (j-c-c) [inline tests/test_basis_utils.py:23]: Should we also test the branch where `m < 0`?

**Review by j-c-c — CHANGES_REQUESTED** [review]
Looks great! Just one thing.

File: tests/test_basis_utils.py (hunk 1 above)
Comment (garrettwrong) [reply tests/test_basis_utils.py:23]: We don't use, but yes, I had a typo there... fixed.

**Review by j-c-c — APPROVED** [review]
Looks good!

**Review by janden — APPROVED** [review]
Perfect. Good riddance!

**Pull request merged**